addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -106,7 +106,11 @@ def _get_partner_party_identification_vals_list(self, partner):
         """ Override to include/update values specific to ZATCA's UBL 2.1 specs """
         return [{
             'id_attrs': {'schemeID': partner.l10n_sa_additional_identification_scheme},
-            'id': partner.l10n_sa_additional_identification_number if partner.l10n_sa_additional_identification_scheme != 'TIN' else partner.vat
+            'id': (
+                partner.l10n_sa_additional_identification_number
+                if partner.l10n_sa_additional_identification_scheme != 'TIN' and partner.country_code == 'SA'
+                else partner.vat
+            ),
```

Comment:
I think either we should either add a comment explaining shortly the usecase here or we should add a full test for that UAE partner.